astropy/table/tests/test_table.py

```diff
@@ -909,6 +909,55 @@ def test_remove_row_preserves_meta(self, table_types):
         assert self.t.dtype == np.dtype([(str('a'), 'int'),
                                          (str('b'), 'int')])
 
+    def test_remove_masked_rows_1(self, table_types):
+        self._setup(table_types)
+        self.t.add_column(self.b)
+        self.t.add_column(self.c)
+        self.t.remove_masked_rows()
+        assert self.t.colnames == ['a', 'b', 'c']
+        assert np.all([self.t['a'] == np.array([1, 2, 3])])
+        assert np.all([self.t['b'] == np.array([4, 5, 6])])
+        assert np.all([self.t['c'] == np.array([7, 8, 9])])
+
+    def test_remove_masked_rows_2(self):
+        t = table.Table.read([' a, b, c',
+                            ' , 2.0, 3.11',
+                            '1.34, 2.12, 3.1',
+                            '89.12, 33.24, 11.22',
+                            '12, 1.17, ',
+                            ], format="ascii", delimiter=",") 
+        assert t.colnames == ['a', 'b', 'c']
+        t.remove_masked_rows()
+        assert t.colnames == ['a', 'b', 'c']
+        assert len(t) == 3
+        assert np.all([t['a'] == np.array([1.34, 89.12])])
+        assert np.all( [t['b'] == np.array([2.12, 33.24])])
+        assert np.all([t['c'] == np.array([3.1, 11.22])])
+
+    def test_remove_masked_rows_2(self):
```

Comment:
What is this test doing different from the previous?  In fact I just noticed this has the same name as the previous one, so that one was never being run.  This explains why the `assert len(t) == 3` was not failing.